lib/playerlist.js

```diff
@@ -0,0 +1,152 @@
+const { LitElement, html, css } = require('lit-element')
+
+class PlayerList extends LitElement {
+  static get styles () {
+    return css`
+        .playerlist-container {
+          position: absolute;
+          background-color: rgba(0, 0, 0, 0.5);
+          left: 50%;
+          top: 10%;
+          transform: translateX(-50%);
+          border: 2px solid rgba(0, 0, 0, 0.8);
+          padding: 10px;
+          min-width: 8%;
+        }
+        
+        .playerlist-entry {
+          overflow: hidden;
+          color: white;
+          font-size: 20px;
+          margin: 0px;
+          line-height: 100%;
+          text-shadow: 2px 2px 0px #3f3f3f;
+          font-family: mojangles, minecraft, monospace;
+          width: 100%;
+        }
+
+        .plist-active-player {
+          color: rgb(42, 204, 237);
+        }
+        
+        .plist-ping-container {
+          text-align: right;
+          float: right;
+          padding-left: 20px;
+        }
+
+        .plist-ping-value {
+          color: rgb(114, 255, 114);
+          float: left;
+          margin: 0px;
+        }
+
+        .plist-ping-label {
+          color: white;
+          float: right;
+          margin: 0px;
+        }
+    `
+  }
+
+  render () {
+    return html`
+      <div id="playerlist-container" class="playerlist-container" style="display: none;">
+      </div>
+    `
+  }
+
+  init (bot) {
+    const playerList = this.shadowRoot.querySelector('#playerlist-container')
+
+    Object.values(bot.players).forEach(player => {
+      addPlayer(player)
+    })
+
+    this.isOpen = false
+
+    const self = this
+
+    document.onkeydown = function (e) {
+      e = e || window.event
+      if (e.key === 'Tab') {
+        showList(true)
+        e.preventDefault()
+      }
+    }
+
+    document.onkeyup = function (e) {
+      if (!self.isOpen) return
+      e = e || window.event
+      if (e.key === 'Tab') {
+        showList(false)
+        e.preventDefault()
+      }
+    }
+
+    function showList (shouldShow = true) {
+      if (shouldShow) {
+        playerList.style.display = 'block'
+      } else {
+        playerList.style.display = 'none'
+      }
+
+      self.isOpen = shouldShow
+    }
+
+    function addPlayer (player) {
+      const playerEntry = document.createElement('div') // There is likely a better way to handle this CSS-wise
+
+      const playerPingContainer = document.createElement('div')
```

Comment:
it's ok but just fyi, in lit element we have a templating language that is fast, so you can just do a loop in render function to achieve the same thing with nicer code

Reply:
I might look into lit element's features and make it nicer later bc the way it looks rn really bothers me :)
I just blindly followed the way you adapted other parts this time tbh

Reply:
I didn't adapt other parts very well, just stuff things in components so it works, but it's not the idiomatic way
https://github.com/PrismarineJS/prismarine-web-client/blob/master/lib/loading_screen.js is a bit better

lit element has a nice tutorial https://lit-element.polymer-project.org/guide/templates